cpp/src/arrow/ipc/options.h

```diff
@@ -161,6 +173,16 @@ struct ARROW_EXPORT IpcReadOptions {
   /// RecordBatchStreamReader and StreamDecoder classes.
   bool ensure_native_endian = true;
 
+  /// \brief How to align data if mis-aligned
+  ///
+  /// Data is copied to aligned memory locations allocated via the
+  /// MemoryPool configured as \ref arrow::ipc::IpcReadOptions::memory_pool.
+  /// Some use cases might require data to have a specific alignment, for example,
+  /// for the data buffer of an Int32 array to be aligned on a 4-byte boundary.
+  ///
+  /// Default (kAnyAlignment) keeps the alignment as is, so no copy of data occurs.
+  Alignment ensure_alignment = Alignment::kAnyAlignment;
```

Comment:
So we only allow to pass one of the enum values above, right?

Reply:
Yes, no combination of enum values expected.

Reply:
My question was more about allowing arbitrary alignment values. But I think it's fine anyway.

Reply:
I see, technically, you could use any positive integer for alignment, it is not limited to the existing constants.